util/dvsim/dvsim.py

```diff
@@ -109,6 +110,74 @@ def sigint_handler(signal_received, frame):
     exit(1)
 
 
+def pick_dump_format(fmts):
```

Comment:
We need to add an overarching validation script to check for the supplied args in any case. 

This file needs to remain as agnostic to what flow is being run as possible, since dvsim is used to run lint, fpv and synthesis as well. 

For the time being, would you please move the verdi detection to the hjson / tcl file, instead? You can do it like this:

In common_sim_cfg.hjson:
```
  has_verdi: "{eval_cmd} command -v verdi > /dev/null 2>&1 && echo 1 || echo 0"
  exports: [
    TOOL: {tool} # use this in the tcl file to switch between vcs/xcelium/etc
    HAS_VERDI: {has_verdi} # use this in the tcl file to default to fsdb/vpd
    TOOL_SRCS_DIR: {tool_srcs_dir} # use this in the tcl file to import the common waves.tcl
    ...
  ]
```

Reply:
I've just re-looked at the code I wrote and it definitely didn't do the right thing (oops: it didn't work if you didn't specify a tool on the command line). I've moved the dump format auto-detection into `SimCfg.py`, which I think makes more sense. Sadly, the patch no longer prints an error if you do something like `--tool verible --dump fsdb`, but we didn't have a check for that before either, so I guess it doesn't really matter.

Does the result look ok to you? I'd much rather not put program logic (searching for `verdi` on $PATH or picking a default dumping format for each simulator) in the hjson files: it's much clearer in the Python, I think. Note that your proposal still doesn't show how to choose a default dumping format differently between VCS and Xcelium when Verdi isn't available. Obviously, you *can* hack this into the TCL, but it's much easier in Python!